### PR TITLE
Run tests on push to origin

### LINKED ISSUE #2790: Run tests on push to remote
We should be able to run our `lint/tsc/test` commands locally before committing and pushing code changes.

## AC
- [x] `npm run test` should be executed when pushing to remote git origin
- [x] husky library updated to the latest version
- [x] use `lint-staged` to lint code before committing

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
I want to merge this change because it fixes #2790 

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: .husky/pre-push [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
```
Comment (krzysztofzuraw) [inline .husky/pre-push:3]: **question**: should I add `npm run lint` as well here?

Comment (Droniu) [reply .husky/pre-push:3]: If it takes more than 5 seconds I would say no

File: package.json [hunk 2]
```diff
@@ -293,11 +299,12 @@
     "cy:run:stable:parallel": "cypress run --record --env grepTags=@critical  --parallel --tag Stable",
     "test": "jest src/",
     "test:watch": "jest --watch src/",
-    "lint": "npx eslint \"src/**/*.@(tsx|ts|jsx|js)\" --fix ; npx prettier --check \"src/**/*.@(tsx|ts|jsx|js)\" --write",
+    "lint": "eslint \"src/**/*.@(tsx|ts|jsx|js)\" --fix",
```
Comment (Droniu) [inline package.json:302]: **question**: Why is prettier removed from here?

**Review by Droniu — APPROVED** [review]

File: .husky/pre-push (hunk 1 above)
Comment (krzysztofzuraw) [reply .husky/pre-push:3]: It took `6.6262s` on my machine with m1 😄

**Review by andrzejewsky — APPROVED** [review]

**Review by Droniu — DISMISSED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Pull request merged**